src/main/java/org/spongepowered/api/entity/Item.java

```diff
@@ -25,17 +25,88 @@
 
 package org.spongepowered.api.entity;
 
+import com.google.common.base.Optional;
+import org.spongepowered.api.entity.player.Player;
 import org.spongepowered.api.item.inventory.ItemStack;
 
+import javax.annotation.Nullable;
+
 /**
  * Represents an Item entity.
  */
 public interface Item extends Entity {
 
     /**
-     * Get the {@link ItemStack} that this entity represents.
+     * Get the {@link ItemStack} that this Item represents.
      *
      * @return The represented {@link ItemStack}
      */
     ItemStack getItemStack();
+
+    /**
+     * Gets the number of ticks remaining until this Item can be picked up,
+     * or -1 if this Item has an infinite pickup delay.
+     *
+     * @return The number of ticks remaining, or -1
+     */
+    int getPickupDelay();
+
+    /**
+     * Sets the number of ticks remaining until this Item can be picked up.
+     *
+     * <p>If this Item currently has an infinite pickup delay, the infinite
+     * pickup delay will be removed, and delay will be set instead.</p>
+     *
+     * @param delay The number of ticks remaining
+     */
+    void setPickupDelay(int delay);
+
+    /**
+     * Sets whether this Item has an infinite pickup delay
+     *
+     * @param infinite Whether this Item has an infinite pickup delay
+     */
+    void setInfinitePickupDelay(boolean infinite);
```

Comment:
You could create a new class, or use implementation dependent "magic constants" instead, reference the constant, and let the implementation define what the constant is.

Also what does setInfinitePickupDelay(false) mean? what would the delay be set to?

IMO you could have it not accept anything, and have setPickupDelayInfinite() (naming is not verbal, but would allow autocomplete to suggest it when you type setPickupDelay.